hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/producer/kafka/KafkaMessagesAsyncProducerTest.groovy

```diff
@@ -0,0 +1,79 @@
+package pl.allegro.tech.hermes.frontend.producer.kafka
+
+import pl.allegro.tech.hermes.frontend.metric.CachedTopic
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessageProducer
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingResults
+import pl.allegro.tech.hermes.frontend.publishing.PublishingCallback
+import pl.allegro.tech.hermes.frontend.publishing.message.JsonMessage
+import pl.allegro.tech.hermes.frontend.publishing.message.Message
+import spock.lang.Shared
+import spock.lang.Specification
+
+import static com.google.common.base.Charsets.UTF_8
+import static pl.allegro.tech.hermes.frontend.server.CachedTopicsTestHelper.cachedTopic
+
+class KafkaMessagesAsyncProducerTest extends Specification {
+
+    @Shared
+    CachedTopic cachedTopic = cachedTopic("group.test")
+
+    def "should successfully publish messages async if publishing single message on broker is available"() {
+        given:
+        List<Message> jsonMessages = (0..<4).collect {
+            new JsonMessage(it.toString(), "{\"data\":\"json\"}".getBytes(UTF_8), 1L, null)
```

Comment:
maybe we could create method for creating JsonMessage ? This code is repeated few times.The only argument which changes is the ID of the message

Reply:
fixed